### PR TITLE
[server] Multiples fixes for wms server

### PR DESCRIPTION
**Pull request by tcoupin**
## Description
This PR includes multiples fixes:
- **Fix group wms dialog**: on "set group wms data", you can set short name, title and abstract. On next time, title contains abstract and abstract is empty. 
- Fix group dialog for qgz project: in global wms configuration, you can exclude layers but it was not working for qgz project file because group dialog was opening project file and put it in a domdocument. group dialog detects now if the project file is zipped.
- **Use group short name for GetMap operation**: if a group has a short name, it appears in GetCap but cannot be in GetMap.
- **Check if layer is queryable for GetFeatureInfo**

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Review by rldhont — COMMENTED** [review]
Tests needed

**Comment by pblottiere** [comment]
Hi @tcoupin,

Thanks a lot for your contribution, you spotted some serious issues here!

I am surprised that OGC tests for WMS 1.3.0 don't check the `LayerNotQueryable` Service exception for the `GetFeatureInfo` request :-/ (it definitely should).

Before merging your PR, I'll run OGC tests to check that everything is still green.

File: src/server/services/wms/qgswmsrenderer.cpp [hunk 1]
```diff
@@ -2411,7 +2417,12 @@ namespace QgsWms
     {
       if ( child->nodeType() == QgsLayerTreeNode::NodeGroup )
       {
-        initLayerGroupsRecursive( static_cast<const QgsLayerTreeGroup *>( child ), child->name() );
+        QString shortName = child->customProperty( QStringLiteral( "wmsShortName" ) ).toString();
+        QString name = child->name();
+        if ( !shortName.isEmpty() )
+          initLayerGroupsRecursive( static_cast<const QgsLayerTreeGroup *>( child ), shortName );
+        else
+          initLayerGroupsRecursive( static_cast<const QgsLayerTreeGroup *>( child ), name );
```
Comment (pblottiere) [inline src/server/services/wms/qgswmsrenderer.cpp]: It looks good, but I would have preferred something shorter :):

```
QString name = child->customProperty( QStringLiteral( "wmsShortName" ) ).toString();

if ( name.isEmpty() )
  name = child->name();

initLayerGroupsRecursive( static_cast<const QgsLayerTreeGroup *>( child ), name );
```

File: src/app/qgsprojectlayergroupdialog.cpp [hunk 2]
```diff
@@ -168,10 +169,40 @@ void QgsProjectLayerGroupDialog::changeProjectFile()
...
+    {
+      return;
+    }
+
+    if ( !projectDom.setContent( &projectFile ) )
+    {
+      return;
+    }
+  } else {
+    if ( !projectDom.setContent( &projectFile ) )
```
Comment (pblottiere) [inline src/app/qgsprojectlayergroupdialog.cpp]: Unneeded imbrication.

File: src/server/services/wms/qgswmsrenderer.cpp (hunk 1 above)
Comment (tcoupin) [reply src/server/services/wms/qgswmsrenderer.cpp]: done

File: src/app/qgsprojectlayergroupdialog.cpp (hunk 2 above)
Comment (tcoupin) [reply src/app/qgsprojectlayergroupdialog.cpp]: done

**Comment by tcoupin** [comment]
Hi,
@pblottiere, request refactoring are done.LayerNotQueryable is throw only if query_layer is empty.
@rldhont, I can't find existing unit tests to improve. Can you help me ?

**Comment by rldhont** [comment]
Hi @tcoupin,

Firstly, you can update one of the project used in **test_qgsserver.py** and **test_qgsserver_*.py** by adding a **wmsAbstract** to a group and check the capabilities. You find the project in the directories **tests/testdata/qgis_server/** and **tests/testdata/qgis_server_*/**.

Secondly, take a look at the **tests/src/python/test_qgsserver_wms_getfeatureinfo.py** and a test querying a **LayerNotQueryable**.

**Review by rldhont — CHANGES_REQUESTED** [review]
```
Traceback (most recent call last):
  File "/root/QGIS/tests/src/python/test_qgsserver_wms_getfeatureinfo.py", line 143, in testGetFeatureInfo
    'wms_getfeatureinfo_invalid_query_layers')
  File "/root/QGIS/tests/src/python/test_qgsserver_wms.py", line 69, in wms_request_compare
    self.assertXMLEqual(response, expected, msg=msg)
  File "/root/QGIS/tests/src/python/test_qgsserver.py", line 92, in assertXMLEqual
    response_element_content[0], msg=msg + "\nContent mismatch on line %s: %s != %s" % (line_no, expected_line, response_line))
AssertionError: b"Layer 'InvalidLayer' not found" != b"Layer 'InvalidLayer' is not queryable" : request https://www.qgis.org/?MAP=/root/QGIS/tests/testdata/qgis_server/test_project.qgs&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetFeatureInfo&layers=testlayer%20%C3%A8%C3%A9&INFO_FORMAT=text%2Fxml&width=600&height=400&srs=EPSG%3A3857&query_layers=InvalidLayer&FEATURE_COUNT=10&FILTER_GEOM=POLYGON((8.2035381 44.901459,8.2035562 44.901459,8.2035562 44.901418,8.2035381 44.901418,8.2035381 44.901459)) failed.
Query: GetFeatureInfo
Expected file: /root/QGIS/tests/testdata/qgis_server/wms_getfeatureinfo_invalid_query_layers.txt
Response:
*****
Content-Type: text/xml; charset=utf-8
<ServiceExceptionReport xmlns="http://www.opengis.net/ogc" version="1.3.0">
 <ServiceException code="LayerNotQueryable">Layer 'InvalidLayer' is not queryable</ServiceException>
</ServiceExceptionReport>
Content mismatch on line 5: b'<ServiceException code="LayerNotDefined">Layer \'InvalidLayer\' not found</ServiceException>' != b'<ServiceException code="LayerNotQueryable">Layer \'InvalidLayer\' is not queryable</ServiceException>'
```

You need to update **tests/testdata/qgis_server/wms_getfeatureinfo_invalid_query_layers.txt**

**Comment by elpaso** [comment]
Just a generic advice here, because we've been problems in the past with using some wierd characters in strings, it always a good idea to put some wierd unicode chars in field names, field values and other strings to make sure they are interpreted correctly in the output.

**Comment by tcoupin** [comment]
I did unit tests for getcap and getmap.

For getFI, group are specific cases of non queryable layer.

**Comment by tcoupin** [comment]
I did some changes to fix unit tests.
Qgs files have been made using gui and there are a lot of changes between v2.99 and v3.2 qgs format... 

I will be on vacation the next weeks so I will see travis status tomorow and make the requested changes at the end of August

**Pull request merged**